File: Aryan's Satellite Control/claude.human.py
import sys
from bisect import bisect_left, bisect_right

class Fenwick:
  def __init__(self, n):
    self.n = n
    self.bit = [0] * (n + 1)

  def add(self, i, value):
    i += 1
    while i <= self.n:
      self.bit[i] += value
      i += i & -i

  def sum(self, i):
    total = 0
    i += 1
    while i:
      total += self.bit[i]
      i -= i & -i
    return total

  def kth(self, k):
    idx = 0
    step = 1 << (self.n.bit_length() - 1)
    while step:
      nxt = idx + step
      if nxt <= self.n and self.bit[nxt] < k:
        idx = nxt
        k -= self.bit[nxt]
      step >>= 1
    return idx

input = sys.stdin.readline

for _ in range(int(input())):
  n, k = map(int, input().split())
  jobs = [tuple(map(int, input().split())) for _ in range(n)]

  jobs.sort(key=lambda x: x[1])
  ends = sorted(set([0] + [r for _, r in jobs]))

  fw = Fenwick(len(ends))
  fw.add(bisect_left(ends, 0), k)

  ans = 0

  for left, right in jobs:
    idx = bisect_right(ends, left) - 1
    if idx < 0 or fw.sum(idx) == 0:
      continue

    count = fw.sum(idx)
    chosen = fw.kth(count)
    fw.add(chosen, -1)
    fw.add(bisect_left(ends, right), 1)
    ans += 1

  print(ans)

File: Aryan's Satellite Control/claude.human.cpp
#include <bits/stdc++.h>
using namespace std;

struct Fenwick {
    int n;
    vector<int> bit;

    Fenwick(int n) : n(n), bit(n + 1) {}

    void add(int i, int v) {
        for (i++; i <= n; i += i & -i) bit[i] += v;
    }

    int sum(int i) {
        int res = 0;
        for (i++; i; i -= i & -i) res += bit[i];
        return res;
    }

    int kth(int k) {
        int pos = 0;
        int pw = 1;
        while ((pw << 1) <= n) pw <<= 1;

        for (; pw; pw >>= 1) {
            int nxt = pos + pw;
            if (nxt <= n && bit[nxt] < k) {
                pos = nxt;
                k -= bit[nxt];
            }
        }
        return pos;
    }
};

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);

    int T;
    cin >> T;

    while (T--) {
        int n, k;
        cin >> n >> k;

        vector<pair<long long, long long>> a(n);
        vector<long long> coords;

        for (auto &[l, r] : a) {
            cin >> l >> r;
            coords.push_back(r);
        }

        sort(a.begin(), a.end(), [](auto &x, auto &y) {
            return x.second < y.second;
        });

        coords.push_back(0);
        sort(coords.begin(), coords.end());
        coords.erase(unique(coords.begin(), coords.end()), coords.end());

        Fenwick fw(coords.size());
        fw.add(0, k);

        int ans = 0;

        for (auto [l, r] : a) {
            int idx = upper_bound(coords.begin(), coords.end(), l) - coords.begin() - 1;
            if (idx < 0) continue;

            int available = fw.sum(idx);
            if (available == 0) continue;

            int chosen = fw.kth(available);
            fw.add(chosen, -1);

            int ridx = lower_bound(coords.begin(), coords.end(), r) - coords.begin();
            fw.add(ridx, 1);
            ++ans;
        }

        cout << ans << '\n';
    }
}
